baselines/FedPer/README.md

```diff
@@ -0,0 +1,135 @@
+
+****Datasets:**** : CIFAR10 from PyTorch's Torchvision and FLICKR-AES. FLICKR-AES was proposed as dataset in _Personalized Image Aesthetics_ (Ren et al., 2017) and can be downloaded using a link provided on thier [GitHub](https://github.com/alanspike/personalizedImageAesthetics). One must first download FLICKR-AES-001.zip (5.76GB), extract all inside and place in baseline/FedPer/datasets. To this location, also download the other 2 related files: (1) FLICKR-AES_image_labeled_by_each_worker.csv, and (2) FLICKR-AES_image_score.txt. Current repository supports CIFAR100 but it hasn't been used to reproduce figures. 
+
+****Hardware Setup:**** : Experiments have been carried out on GPU. 2 different computers managed to run experiments: 
+
+- GeForce RTX 3080 16GB
+- GeForce RTX 4090 24GB
+
+****Contributors:**** : William Lindskog
+
+
+## Experimental Setup
+
+****Task:**** : Image Classification
+
+****Model:**** : This directory implements 2 models:
+
+- ResNet34 which can be imported directly (after having installed the packages) from PyTorch, using `from torchvision.models import resnet34 
+- MobileNet-v1
+
+Please see how models are implemented using a so called model_manager and model_split class since FedPer uses head and base layers in a neural network. These classes are defined in the models.py file and thereafter called when building new models in the directory /implemented_models. Please, extend and add new models as you wish. 
+
+****Dataset:**** : CIFAR10, FLICKR-AES. CIFAR10 will be partitioned based on number of classes for data that each client shall recieve e.g. 4 allocated classes could be [1, 3, 5, 9]. FLICKR-AES is an unbalanced dataset, so there we only apply random sampling. 
+
+****Training Hyperparameters:**** : The hyperparameters can be found in conf/base.yaml file which is the configuration file for the main script. 
+
+| Description | Default Value |
+| ----------- | ----- |
+| num_clients | 10 |
+| clients per round | 10 |
+| number of rounds | 50 |
+| client resources | {'num_cpus': 32, 'num_gpus': 1 }|
+| learning_rate | 0.01 |
+| batch_size | 128 |
+| optimizer | SGD |
+| algorithm | fedavg|
+
+
+## Environment Setup
+
+To construct the Python environment follow these steps:
+
+```bash
+# install the base Poetry environment
+poetry install
+
+# activate the environment
+poetry shell
+
+# install PyTorch with GPU support. Please note this baseline is very lightweight so it can run fine on a CPU.
+pip install torch==2.0.1+cu117 torchvision==0.15.2+cu117 
+```
+## Running the Experiments
+```bash
+python -m FedPer.main # this will run using the default settings in the `conf/base.yaml`    
+```
+
+To reproduce figures:
+```bash
+# make FedPer/run_scripts.sh executable
+chmod u+x FedPer/run_scripts.sh
+# uncomment lines in script that you want to run, then  
+bash FedPer/run_scripts.sh
```

Comment:
You have a nice `--multirun` command a few lines below. What is the motivation for running a bash script instead of a multirun? I think at least a few of the blocks commands you run in the script can be replaces with a single multi-run command. Up to you if you think it's worth doing this.